include/gnc/orbit_controller.hpp

```diff
@@ -0,0 +1,139 @@
+//
+// MIT License
+//
+// Copyright (c) 2020 Pathfinder for Autonomous Navigation (PAN)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+//
+
+/** @file gnc/attitude_estimator.hpp
+ *  @author Kyle Krol
+ */
+
+/** @defgroup orbit_controller Orbit Controller
+ *  @brief Defines the interface for the orbit controller.
+ * 
+ *  More details to come...
+ */
+
+#ifndef GNC_ORBIT_CONTROLLER_HPP_
+#define GNC_ORBIT_CONTROLLER_HPP_
+
+#include <lin/core.hpp>
+
+namespace gnc {
+
+/** @brief Contains the internal state of an orbit controller.
+ * 
+ *  The internal state only includes a timestamp of the last recommended firing in
+ *  seconds since the PAN epoch. This is to avoid accidently suggesting two
+ *  firings due to a single firing node.
+ * 
+ *  The rest of the state struct serves as a calculation buffer to the controller.
+ *  This is to aid with reducing the stack size required to run it.
+ *  
+ *  @sa OrbitControllerData
+ *  @sa OrbitActuation
+ * 
+ *  @ingroup orbit_controller
+ */
+struct OrbitControllerState {
+  double t_last_firing; //!< Last firing's timestamp since the PAN epoch (s).
```

Comment:
I'd recommend using `int64_t` nanoseconds since GPS epoch for better compatibility with `orb`

Reply:
Initialized to 0.

Reply:
Initialized to zero or to it's min value? My intention with starting it at NaN was it wouldn't prevent any firing from being scheduled because it was too close to the next one.